integration-tests/scripts/wait-for-release.sh

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env bash
+#
+# Summary:
+#   Monitors a Kubernetes Release custom resource until it reaches a terminal state
+#   (Succeeded or Failed). It periodically fetches the Release status, displays
+#   a summary of its various processing stages (Tenant Collectors, Managed Collectors,
+#   Tenant Pipeline, Managed Pipeline, Final Pipeline), and provides console URLs
+#   for associated PipelineRuns. If the Release fails, it attempts to fetch and
+#   display logs from the relevant PipelineRuns.
+#
+# Parameters:
+#   None.
+#
+# Environment Variables:
+#   RELEASE_NAME      - The name of the Release custom resource to monitor. Required.
+#   RELEASE_NAMESPACE - The Kubernetes namespace where the Release resource exists. Required.
+#   CONSOLEURL        - (Derived by the script) The base URL for the OpenShift console,
+#                       used to construct links to PipelineRuns and Releases.
+#                       It's fetched from the 'pipelines-as-code' ConfigMap.
+#   APPLICATION       - (Derived by the script) The name of the application associated
+#                       with the Release, extracted from the Release's labels.
+#
+# Dependencies:
+#   kubectl, jq, tkn (for fetching PipelineRun logs), oc (for fetching CONSOLEURL).
+
+function getPipelinerunFromStatus() { # args are json, statusSection
+  json=$1
+  statusSection=$2
+
+  filter=".status."$statusSection".pipelineRun"
+  PLR=$(jq -r $filter <<< "${json}")
+  echo ${PLR//null/}
+}
+
+function getConsoleLogForRelease() { # args are application, ns, name
+  app=$1
+  ns=$2
+  name=$3
+  releaseUrl="${CONSOLEURL}ns/${ns}/applications/${app}/releases/${name}"
+  echo "${releaseUrl}"
+}
+
+function getConsoleLogFromPLR() { # args are json, statusSection
```

Comment:
Is there any good way to get the metadata of the pipelineRun? I know we can get pipelineRun logs from splunk.  Is there any convenient way to get the pipelineRun yaml file since it's also helpful for debugging?

Reply:
we can look at getting the metadata and actual PLR yaml in a subsequent PR. Sound ok?